src/conditions/datadog_search.rs

```diff
@@ -27,215 +26,297 @@ impl_generate_config_from_default!(DatadogSearchConfig);
+                }
+                // Unbounded upper.
+                (_, ComparisonValue::Unbounded) => {
+                    let op = if *lower_inclusive {
+                        Comparison::Gte
+                    } else {
+                        Comparison::Gt
+                    };
 
-                array_match(field, move |values| {
-                    values.contains(&Value::Bytes(Bytes::copy_from_slice(to_match.as_bytes())))
-                })
-            }
-            // Individual tags are compared by element key:value.
-            Field::Tag(tag) => {
-                let value_bytes = Value::Bytes(format!("{}:{}", tag, to_match).into());
+                    compare(field, op, lower, log)
+                }
+                // Definitive range.
+                _ => {
+                    let lower_op = if *lower_inclusive {
+                        Comparison::Gte
+                    } else {
+                        Comparison::Gt
+                    };
 
-                array_match("tags", move |values| values.contains(&value_bytes))
-            }
-            // Everything else is matched by string equality.
-            Field::Reserved(field) | Field::Facet(field) => {
-                let to_match = to_match.to_owned();
+                    let upper_op = if *upper_inclusive {
+                        Comparison::Lte
+                    } else {
+                        Comparison::Lt
+                    };
 
-                string_match(field, move |value| value == to_match)
+                    compare(field, lower_op, lower, log) && compare(field, upper_op, upper, log)
+                }
             }
         }
+        Op::Not(matcher) => !EventFilter::run(matcher, log),
+        Op::Nested(matcher) => EventFilter::run(matcher, log),
+    }
+}
+
+fn exists(field: &Field, log: &LogEvent) -> bool {
+    match field {
+        Field::Tag(tag) => match log.get("tags") {
+            Some(Value::Array(values)) => values
+                .iter()
+                .filter_map(|value| {
+                    if let Value::Bytes(bytes) = value {
+                        std::str::from_utf8(bytes).ok()
+                    } else {
+                        None
+                    }
+                })
+                .any(|value| {
+                    value == tag
+                        || (value.starts_with(tag) && value.chars().nth(tag.len()) == Some(':'))
+                }),
+            _ => false,
+        },
+        // Literal field 'tags' needs to be compared by key.
+        Field::Reserved(field) if field == "tags" => match log.get("tags") {
+            Some(Value::Array(values)) => values
+                .iter()
+                .filter_map(|value| {
+                    if let Value::Bytes(bytes) = value {
+                        std::str::from_utf8(bytes).ok()
+                    } else {
+                        None
+                    }
```

Comment:
`match`

Reply:
Is the benefit just that it's one line shorter? Using `if let` to match a single variant seems pretty idiomatic.